test.js

```diff
@@ -1,83 +1,36 @@
 'use strict'
 
-var redis = require('./')
 var test = require('tape').test
-var abstractTests = require('mqemitter/abstractTest.js')
 
-abstractTests({
-  builder: redis,
-  test: test
-})
-
-function noop () {}
-
-test('actual unsubscribe from Redis', function (t) {
-  var e = redis()
-
-  e.subConn.on('message', function (topic, message) {
-    t.fail('the message should not be emitted')
-  })
-
-  e.on('hello', noop)
-  e.removeListener('hello', noop)
-  e.emit({ topic: 'hello' }, function () {
-    e.close(function () {
-      t.end()
-    })
-  })
-})
-
-test('ioredis connect event', function (t) {
-  var e = redis()
-
-  var subConnectEventReceived = false
-  var pubConnectEventReceived = false
-
-  e.state.on('pubConnect', function () {
-    pubConnectEventReceived = true
-    newConnectionEvent()
-  })
-
-  e.state.on('subConnect', function () {
-    subConnectEventReceived = true
-    newConnectionEvent()
+function runTests (builder, isCluster) {
+  require('mqemitter/abstractTest.js')({
+    builder: builder,
+    isCluster: isCluster || false,
+    test: test
   })
-
-  function newConnectionEvent () {
-    if (subConnectEventReceived && pubConnectEventReceived) {
-      e.close(function () {
-        t.end()
-      })
-    }
-  }
-})
-
-test('ioredis error event', function (t) {
-  var e = redis({host: '127'})
-
-  t.plan(1)
-
-  e.state.once('error', function (err) {
-    t.deepEqual(err.message.substr(0, 7), 'connect')
-    e.close(function () {
-      t.end()
-    })
-  })
-})
-
-test('topic pattern adapter', function (t) {
-  var e = redis()
-
-  var mqttTopic = 'rooms/+/devices/+/status'
-  var expectedRedisPattern = 'rooms/*/devices/*/status'
-
-  var subTopic = e._subTopic(mqttTopic)
-
-  t.plan(1)
-
-  t.deepEqual(subTopic, expectedRedisPattern)
-
-  e.close(function () {
-    t.end()
+  require('./redisTest.js')({
+    builder: builder,
+    isCluster: isCluster || false,
+    test: test
   })
-})
+}
+
+runTests(require('./'))
```

Comment:
can you move this require to the top of the file, and also the next one?

Reply:
sure. here you are ab5956f6ae2736980a9b748bb0ea3043344c372c